File: src/main/java/com/quatico/workshops/webcomponents/message/Message.java
package com.quatico.workshops.webcomponents.message;

public class Message {
    private String text;
    private String signature;

    public Message() {
        this("Message");
    }

    public Message(String text) {
        this.text = text;
    }


    public Message sign(String secret, EncodeFn encodeFn) {
        this.signature = encodeFn.apply(secret, this.text);
        return this;
    }

    public String getSignature() {
        return this.signature;
    }

    public String getText() {
        return this.text;
    }
}

File: src/main/typescript/solution1/Message.ts
import { EncodeFn } from "./EncodeFn";

export class Message {
    public readonly text: string;
    private signature?: string;

    constructor(text?: string) {
        this.text = text ?? "Message";
    }

    public sign(secret: string, encodeFn: EncodeFn): Message {
        this.signature = encodeFn(secret, this.text);
        return this;
    }

    public getSignature(): string | undefined {
        return this.signature;
    }
}
